### PR TITLE
WEBUI: Correctly handle document babies api responses in error cases (BTS-934)

### PR DESCRIPTION
**Pull request by hkernbach**
Backport of: https://github.com/arangodb/arangodb/pull/16820
Please see details there.

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/frontend/js/views/documentView.js [hunk 1]
```diff
@@ -453,21 +453,23 @@
...
           if (error) {
             arangoHelper.arangoError('Error', data.responseJSON.errorMessage);
           } else {
-            // here we need to do an additional error check as PUT API is returning 202 (PUT) with error inside - lol
-            if (data[0].error) {
-              arangoHelper.arangoError('Error', data[0].errorMessage);
+            // Here we need to do an additional error check as PUT API is returning 202 (PUT) with error inside.
+            // Can contain error variable, but can contain errorNum and errorMessage as well (Babies API)
+            const entry = data[0];
+            if (entry.error || (entry.errorNum && entry.errorMessage)) {
```
Comment (adityamukho) [inline js/apps/system/_admin/aardvark/APP/frontend/js/views/documentView.js:463]: Sorry I missed this bit in the main PR, but noticed in the backport. It seems `entry.errorNum` and `entry.errorMessage` are assumed to exist if `entry.error` exists. Is this a correct assumption? If so, we can simply omit the check for `entry.error`.

**Review by adityamukho — COMMENTED** [review]
Please see my comment inline.

File: js/apps/system/_admin/aardvark/APP/frontend/js/views/documentView.js (hunk 1 above)
Comment (hkernbach) [inline js/apps/system/_admin/aardvark/APP/frontend/js/views/documentView.js:463]:
```suggestion
            if (entry.errorNum && entry.errorMessage) {
```

**Review by ajurdak — APPROVED** [review]
LGTM

**Review by adityamukho — APPROVED** [review]
LGTM

**Pull request merged**